pytorch_lightning/trainer/training_loop.py

```diff
@@ -489,6 +489,10 @@ def optimizer_step(self, optimizer, opt_idx, batch_idx, train_step_and_backward_
                 'native PyTorch amp and lbfgs are not compatible.'
                 ' To request, please file a Github issue in PyTorch and tag @mcarilli')
 
+        if not isinstance(optimizer, LightningOptimizer):
+            # wraps into LightingOptimizer only for running step
+            optimizer = LightningOptimizer.to_lightning_optimizer(optimizer, self.trainer)
```

Comment:
Just something to consider, this wraps the optimizer but never unwraps right?

Reply:
It wraps it for `optimizer_step` only but we don't assigned it back to the Trainer, so `self.optimizers` would always be the unwraped ones. Am I missing something ?

Reply:
@tchaton you can move the if statement inside the `to_lightning_optimizer` function to achieve idempotence and avoid code duplication.

Reply:
My mistake! You're right